test/Microsoft.AspNet.Mvc.ModelBinding.Test/ValueProviders/QueryStringValueProviderTest.cs

```diff
@@ -219,17 +219,17 @@ public static IEnumerable<object[]> RegisteredAsMetadataClasses
         public void FilterReturnsItself_ForAnyClassRegisteredAsGenericParam(IValueProviderMetadata metadata)
         {
             // Arrange
-            var valueProvider = new ReadableStringCollectionValueProvider<TestValueProviderMetadata>(_backingStore, null);
+            var valueProvider = new QueryStringValueProvider(_backingStore, null);
 
             // Act
             var result = valueProvider.Filter(metadata);
 
             // Assert
             Assert.NotNull(result);
-            Assert.IsType<ReadableStringCollectionValueProvider<TestValueProviderMetadata>>(result);
+            Assert.IsType<QueryStringValueProvider>(result);
         }
 
-        private class TestValueProviderMetadata : IValueProviderMetadata
+        private class TestValueProviderMetadata : IQueryValueProviderMetadata
```

Comment:
It is used in this test `FilterReturnsItself_ForAnyClassRegisteredAsGenericParam`.